dashboard/src/app/workspaces/workspace-config.service.ts

```diff
@@ -0,0 +1,124 @@
+  /**
+   * Log service.
+   */
+  private $log: ng.ILogService;
+  /**
+   * Promises service.
+   */
+  private $q: ng.IQService;
+  /**
+   * Workspace API interaction.
+   */
+  private cheWorkspace: CheWorkspace;
+  /**
+   * Namespace selector service.
+   */
+  private namespaceSelectorSvc: NamespaceSelectorSvc;
+  /**
+   * Workspace creating service.
+   */
+  private createWorkspaceSvc: CreateWorkspaceSvc;
+  /**
+   * Stack selector service.
+   */
+  private stackSelectorSvc: StackSelectorSvc;
+  /**
+   * Template selector service.
+   */
+  private templateSelectorSvc: TemplateSelectorSvc;
+  /**
+   * Import GitHub project service.
+   */
+  private importGithubProjectService: ImportGithubProjectService;
+
+  /** Default constructor that is using resource injection
+   * @ngInject for Dependency injection
+   */
+  constructor($log: ng.ILogService, $q: ng.IQService, cheWorkspace: CheWorkspace, namespaceSelectorSvc: NamespaceSelectorSvc, createWorkspaceSvc: CreateWorkspaceSvc, stackSelectorSvc: StackSelectorSvc, templateSelectorSvc: TemplateSelectorSvc, importGithubProjectService: ImportGithubProjectService) {
+    this.$log = $log;
+    this.$q = $q;
+    this.cheWorkspace = cheWorkspace;
+    this.namespaceSelectorSvc = namespaceSelectorSvc;
+    this.createWorkspaceSvc = createWorkspaceSvc;
+    this.stackSelectorSvc = stackSelectorSvc;
+    this.templateSelectorSvc = templateSelectorSvc;
+    this.importGithubProjectService = importGithubProjectService;
+  }
+
+  /**
+   * Returns promise to resolve route for workspace creation page.
+   *
+   * @return {ng.IPromise<any>}
+   */
+  resolveCreateWorkspaceRoute(): ng.IPromise<any> {
+    const namespaceIdDefer = this.$q.defer(),
+          workspacesDefer = this.$q.defer();
+
+    // resolve namespace ID, workspaces in namespace
+    this.namespaceSelectorSvc.fetchNamespaces().then((namespaceId: string) => {
+      namespaceIdDefer.resolve(namespaceId);
+
+      if (!namespaceId) {
+        return this.$q.reject();
+      }
+
+      return this.createWorkspaceSvc.fetchWorkspacesByNamespace(namespaceId);
+    }).then((workspaces: Array<che.IWorkspace>) => {
+      workspacesDefer.resolve(workspaces);
+    }, (error: any) => {
+      this.logError(error);
+      workspacesDefer.resolve([]);
+    });
+
+    // resolve GitHub repositories
+    const githubRepositoriesPromise = this.$q.all([this.importGithubProjectService.getOrFetchUserId(), this.importGithubProjectService.getOrFetchOAuthProvider()]).then(() => {
+      return this.importGithubProjectService.askLoad();
+    });
+
+    return this.$q.all([
+      namespaceIdDefer.promise,
+      workspacesDefer.promise,
+      this.stackSelectorSvc.getOrFetchStacks(),
+      this.templateSelectorSvc.getOrFetchTemplates(),
+      githubRepositoriesPromise
+    ]);
+  }
+
+  /**
+   * Prints error message.
+   *
+   * @param {any} error error object or string
+   */
+  private logError(error: any): void {
```

Comment:
I saw using this private logError once for the service. I guess we should have something looks like in separate error service because we have a lot of places with error.data.message checking.